src/main/java/ttt/Board.java

```diff
@@ -0,0 +1,150 @@
+package ttt;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class Board {
+
+    private String[] grid = new String[9];
+    private String status;
+    private String emptyCell;
+    private String winningMark;
+
+    public Board() {
+        this.emptyCell = " ";
+        this.winningMark = null;
+        for (int cell = 0; cell < 9; cell++) {
+            grid[cell] = emptyCell;
+        }
+    }
+
+    public void placeMark(String symbol, int location) {
+        if (!isValidLocation(location)) {
+            this.status = "invalid location";
+        } else if (isCellTaken(location)) {
+            this.status = "taken";
+        } else {
+            grid[location] = symbol;
+            status = "mark placed";
+        }
+    }
+
+    public String getStatus() {
+        return status;
+    }
+
+    public int size() {
+        return grid.length;
+    }
+
+    public String get(int cell) {
+        return grid[cell];
+    }
+
+    public String getWinningMark() {
+        return winningMark;
+    }
+
+    public boolean isAWinFor(String mark) {
+        return isWon() && winningMark.equals(mark);
+    }
+
+    public void clear (int location) {
+        grid[location] = emptyCell;
+    }
+
+    public List<Integer> availableMoves() {
+        List<Integer> moves = new ArrayList<>();
+        for (int cell = 0; cell < grid.length; cell ++) {
+            if (grid[cell].equals(emptyCell)) {
+               moves.add(cell);
+            }
+        }
+        return moves;
+    }
+
+    public List<List<String>> winningPositions() {
+        List<List<String>> positions = new ArrayList<>();
+        positions.addAll(rows());
+        positions.addAll(columns());
+        positions.addAll(diagonals());
+        return positions;
+    }
+
+
+    public boolean isFull() {
+        for (String cell : grid) {
+            if (cell.equals(emptyCell)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public boolean isEmpty() {
+        for (String cell : grid) {
+            if (!cell.equals(emptyCell)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public boolean isDraw() {
+        return (isFull() && !isWon());
+    }
+
+    public boolean isFinished() {
+        return (isDraw() || isWon());
+    }
+
+    public boolean isWon() {
+        for (List<String> selection : winningPositions()) {
+            if (isAWin(selection)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private List<String> getCollection(int first, int middle, int last) {
+        return Arrays.asList(grid[first], grid[middle], grid[last]);
+    }
+
+    private boolean isValidLocation(int location) {
+        return (location <= 8 && location >= 0);
```

Comment:
I would flip these around. Its more natural to read something like
`0 <= location && location <= 8`. Your brain quickly identifies the pattern "between 0 and 8".